packages/inference/src/providers/hf-inference.ts

```diff
@@ -9,39 +9,129 @@
  * and we will tag HF team members.
  *
  * Thanks!
- */
+ */ import type { TextGenerationOutput } from "@huggingface/tasks";
 import { HF_ROUTER_URL } from "../config";
-import type { BodyParams, HeaderParams, ProviderConfig, UrlParams } from "../types";
-
-const makeBaseUrl = (): string => {
-	return `${HF_ROUTER_URL}/hf-inference`;
-};
-
-const makeBody = (params: BodyParams): Record<string, unknown> => {
-	return {
-		...params.args,
-		...(params.chatCompletion ? { model: params.model } : undefined),
-	};
-};
-
-const makeHeaders = (params: HeaderParams): Record<string, string> => {
-	return { Authorization: `Bearer ${params.accessToken}` };
-};
-
-const makeUrl = (params: UrlParams): string => {
-	if (params.task && ["feature-extraction", "sentence-similarity"].includes(params.task)) {
-		/// when deployed on hf-inference, those two tasks are automatically compatible with one another.
-		return `${params.baseUrl}/pipeline/${params.task}/${params.model}`;
-	}
-	if (params.chatCompletion) {
-		return `${params.baseUrl}/models/${params.model}/v1/chat/completions`;
-	}
-	return `${params.baseUrl}/models/${params.model}`;
-};
-
-export const HF_INFERENCE_CONFIG: ProviderConfig = {
-	makeBaseUrl,
-	makeBody,
-	makeHeaders,
-	makeUrl,
-};
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { BodyParams, InferenceTask, UrlParams } from "../types";
+import { toArray } from "../utils/toArray";
+import { TaskProviderHelper } from "./providerHelper";
+interface Base64ImageGeneration {
+	data: Array<{
+		b64_json: string;
+	}>;
+}
+interface OutputUrlImageGeneration {
+	output: string[];
+}
+export class HFInferenceTask extends TaskProviderHelper {
```

Comment:
```suggestion
}

export class HFInferenceTask extends TaskProviderHelper {
```

dont hesitate to be more liberal with newlines